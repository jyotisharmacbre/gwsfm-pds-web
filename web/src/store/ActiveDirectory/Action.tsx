import * as axios from '../../client';
import { ActionType } from './Types/ActionType';
import { Dispatch } from 'redux';

const getADhopSuccess = (response: any) => {
  return {
    type: ActionType.AD_HOP_GET_SUCCESS,
    payload: response
  };
};

const getADhopError = (error: any) => {
  return {
    type: ActionType.AD_HOP_GET_ERROR,
    payload: error
  };
};

const getADpmSuccess = (response: any) => {
  return {
    type: ActionType.AD_PM_GET_SUCCESS,
    payload: response
  };
};

const getADpmError = (error: any) => {
  return {
    type: ActionType.AD_PM_GET_ERROR,
    payload: error
  };
};

const getADpoSuccess = (response: any) => {
  return {
    type: ActionType.AD_PO_GET_SUCCESS,
    payload: response
  };
};

const getADpoError = (error: any) => {
  return {
    type: ActionType.AD_PO_GET_ERROR,
    payload: error
  };
};

export const getADhopOther = (response: any) => {
  console.log(response);
  return {
    type: ActionType.AD_HOP_GET_OTHER,
    payload: response
  };
};

export const getADpmOther = (response: any) => {
  console.log(response);
  return {
    type: ActionType.AD_PM_GET_OTHER,
    payload: response
  };
};

export const getADpoOther = (response: any) => {
  console.log(response);
  return {
    type: ActionType.AD_PO_GET_OTHER,
    payload: response
  };
};

let config = {
  headers: {
    'Content-Type': 'application/json'
  }
};
export const getActiveDirectoryHOP = (searchHOP: string) => {
  return (dispatch: Dispatch) => {
    axios.baseAPI
      .get(
        `/api/DynamicsLookup/getCompanies/${searchHOP}?topCount=50
    `,
        config
      )
      .then(response => {
        dispatch(getADhopSuccess(response.data));
      })
      .catch(error => {
        dispatch(getADhopError(error));
      });
  };
};

export const getActiveDirectoryPM = (searchPM: string) => {
  return (dispatch: Dispatch) => {
    axios.baseAPI
      .get(
        `/api/DynamicsLookup/getCompanies/${searchPM}?topCount=50
    `,
        config
      )
      .then(response => {
        dispatch(getADpmSuccess(response.data));
      })
      .catch(error => {
        dispatch(getADpmError(error));
      });
  };
};

export const getActiveDirectoryPO = (searchPO: string) => {
  return (dispatch: Dispatch) => {
    axios.baseAPI
      .get(
        `/api/DynamicsLookup/getCompanies/${searchPO}?topCount=50
    `,
        config
      )
      .then(response => {
        dispatch(getADpoSuccess(response.data));
      })
      .catch(error => {
        dispatch(getADpoError(error));
      });
  };
};
