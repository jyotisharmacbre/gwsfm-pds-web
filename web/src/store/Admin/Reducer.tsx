import { ActionType } from './Types/ActionType';
import { updateObject } from '../../helpers/utility-helper';
import { initialState } from './InitialState';

const getProjectParametersSuccess = (oldState, action) => {
	return updateObject(oldState, {
		adminDefaultValues: action.payload
	});
};

const getProjectParametersError = (oldState, action) => {
	return updateObject(oldState, {
		error: action.payload
	});
};

const adminReducer = (oldState = initialState, action) => {
	switch (action.type) {
		case ActionType.GET_PROJECT_PARAMETERS_SUCCESS:
			return getProjectParametersSuccess(oldState, action);
		case ActionType.GET_PROJECT_PARAMETERS_ERROR:
			return getProjectParametersError(oldState, action);
		default:
			return oldState;
	}
};

export default adminReducer;
