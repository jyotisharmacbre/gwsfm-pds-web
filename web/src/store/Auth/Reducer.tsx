import { updateObject } from '../../helpers/utility-helper';
import { initialState } from './InitialState';
import { AuthenticationActions } from 'react-aad-msal';

const initializing = (oldState, action) => {
    return oldState;
};

const initialized = (oldState, action) => {
    return oldState;
};

const loginSuccess = (oldState, action) => {
    return oldState;
};

const loginFailed = (oldState, action) => {
    return oldState;
};
const loginError = (oldState, action) => {
    return oldState;
};
const clearError = (oldState, action) => {
    return oldState;
};
const logoutSuccess = (oldState, action) => {
    return updateObject(oldState, {
        token: updateObject(oldState.token, {
            idToken: null,
            expiration: null
        }),
        claims: updateObject(oldState.claims, {
            name: null,
            firstName: null,
            email: null
        })
    });
};
const acquiredIdTokenSuccess = (oldState, action) => {
    return updateObject(oldState, {
        token: updateObject(oldState.token, {
            idToken: action.payload.idToken.rawIdToken,
            expiration: action.payload.idToken.expiration
        }),
        claims: updateObject(oldState.claims, {
            name: action.payload.idToken.claims.name,
            firstName: action.payload.idToken.claims.given_name,
            email: action.payload.idToken.claims.email
        })
    });
};
const acquiredIdTokenError = (oldState, action) => {
    return oldState;
};
const acquiredAccessTokenSuccess = (oldState, action) => {
    return oldState;
};
const acquiredAccessTokenError = (oldState, action) => {
    return oldState;
};
const authenticatedStateChanged = (oldState, action) => {
    return oldState;
};

const authReducer = (oldState = initialState, action) => {
    switch (action.type) {
        case AuthenticationActions.Initializing:
            return initializing(oldState, action);
        case AuthenticationActions.Initialized:
            return initialized(oldState, action);
        case AuthenticationActions.LoginSuccess:
            return loginSuccess(oldState, action);
        case AuthenticationActions.LoginFailed:
            return loginFailed(oldState, action);
        case AuthenticationActions.LoginError:
            return loginError(oldState, action);
        case AuthenticationActions.ClearError:
            return clearError(oldState, action);
        case AuthenticationActions.LogoutSuccess:
            return logoutSuccess(oldState, action);
        case AuthenticationActions.AcquiredIdTokenSuccess:
            return acquiredIdTokenSuccess(oldState, action);
        case AuthenticationActions.AcquiredIdTokenError:
            return acquiredIdTokenError(oldState, action);
        case AuthenticationActions.AcquiredAccessTokenSuccess:
            return acquiredAccessTokenSuccess(oldState, action);
        case AuthenticationActions.AcquiredAccessTokenError:
            return acquiredAccessTokenError(oldState, action);
        case AuthenticationActions.AuthenticatedStateChanged:
            return authenticatedStateChanged(oldState, action);
        default:
            return oldState;
    }
};
export default authReducer;