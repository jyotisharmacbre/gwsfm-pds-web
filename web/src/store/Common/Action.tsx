import { ActionType } from './Types/ActionType';

export const getDefaultState = () => {
	return {
		type: ActionType.DEFAULT
	};
};
