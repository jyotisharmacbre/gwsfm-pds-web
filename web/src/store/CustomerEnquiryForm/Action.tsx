import * as axios from '../../client';
import { store } from '../index';
import { ActionType } from './Types/ActionType';
import { Dispatch } from 'redux';
import { IProjectDetail } from './Types/IProjectDetail';
import { IProject } from './Types/IProject';
import EventType from '../../enums/EventType';

const projectDetailAddSuccess = (response: IProjectDetail, event: EventType) => {
	return {
		type: ActionType.PROJECT_ADD,
		payload: response,
		event: event
	};
};

const projectDetailEditSuccess = (response: IProjectDetail, event: EventType) => {
	return {
		type: ActionType.PROJECT_EDIT_SUCCESS,
		payload: response,
		event: event
	};
};

const projectDetailError = (error: string) => {
	return {
		type: ActionType.PROJECT_ADD_ERROR,
		payload: error
	};
};
const changeProjectStatusState = (status: number) => {
	return {
		type: ActionType.CHANGE_PROJECT_STATUS_STATE,
		payload: status
	};
};

const headers = {
	'Content-Type': 'application/json'
};

export const projectDetailAdd = (data: IProjectDetail, event: EventType) => {
	return (dispatch: Dispatch) => {
		axios.baseAPI
			.post('/api/Projects/customerEnquiry', data, { headers: headers })
			.then((response) => {
				dispatch(projectDetailAddSuccess(response.data, event));
			})
			.catch((error) => {
				dispatch(projectDetailError(error));
			});
	};
};

export const projectDetailEdit = (data: IProjectDetail, event: EventType) => {
	return (dispatch: Dispatch) => {
		axios.baseAPI
			.put('/api/Projects/updatecustomerEnquiry', data, { headers: headers })
			.then((response) => {
				dispatch(projectDetailEditSuccess(response.data, event));
			})
			.catch((error) => {
				dispatch(projectDetailError(error));
			});
	};
};

const getEnquiryOverviewSuccess = (response: IProject) => {
	return {
		type: ActionType.GET_ENQUIRY_OVERVIEW_SUCCESS,
		payload: response
	};
};

const getEnquiryOverviewError = (error: string) => {
	return {
		type: ActionType.GET_ENQUIRY_OVERVIEW_ERROR,
		payload: error
	};
};

export const getEnquiryOverview = (projectId: string) => {
	return (dispatch: Dispatch) => {
		axios.baseAPI
			.get(`api/Projects/${projectId}/enquiryOverview`)
			.then((response) => {
				dispatch(getEnquiryOverviewSuccess(response.data));
			})
			.catch((error) => {
				dispatch(getEnquiryOverviewError(error));
			});
	};
};

const getProjectDetailSuccess = (response: IProjectDetail) => {
	return {
		type: ActionType.GET_PROJECT_DETAIL_SUCCESS,
		payload: response
	};
};

const getProjectDetailError = (error: string) => {
	return {
		type: ActionType.GET_PROJECT_DETAIL_ERROR,
		payload: error
	};
};

export const getProjectDetail = (projectId: string, cache: boolean = true) => {
	return (dispatch: Dispatch) => {
		let storeProjectId = store.getState().project.form.projectId;
		if (cache && storeProjectId && projectId == storeProjectId) dispatch({ type: ActionType.DEFAULT });
		else {
			axios.baseAPI
				.get(`api/Projects/${projectId}/customerEnquiry`)
				.then((response) => {
					dispatch(getProjectDetailSuccess(response.data));
				})
				.catch((error) => {
					dispatch(getProjectDetailError(error));
				});
		}
	};
};

const resetProjectDetailStateDispatch = () => {
	return {
		type: ActionType.RESET_PROJECT_DETAIL_STATE
	};
};

export const resetProjectDetailState = () => {
	return (dispatch: Dispatch) => {
		dispatch(resetProjectDetailStateDispatch());
	};
};

const setProjectIdDispatch = () => {
	return {
		type: ActionType.SET_PROJECT_ID_STATE
	};
};

export const setProjectId = (projectId: string) => {
	return (dispatch: Dispatch) => {
		dispatch(setProjectIdDispatch());
	};
};
export const changeProjectStatus = (status: number) => {
	return (dispatch: Dispatch) => {
		dispatch(changeProjectStatusState(status));
	};
};

const resetProjectDetailStateToInitialDispatch = () => {
	return {
		type: ActionType.RESET_PROJECT_DETAIL_STATE_TO_INITIAL
	};
};

export const resetProjectDetailStateToInitial = () => {
	return (dispatch: Dispatch) => {
		dispatch(resetProjectDetailStateToInitialDispatch());
	};
};

export const updateProjectStatusToInReview = (projectId: string, success, error) => {
	axios.baseAPI
		.put(`/api/Projects/${projectId}/inReview`, null, { headers: headers })
		.then((response) => {
			success(response.data);
		})
		.catch((error) => {
			error(error);
		});
};
