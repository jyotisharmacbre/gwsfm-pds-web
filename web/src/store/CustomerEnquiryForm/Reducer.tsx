import { ActionType } from './Types/ActionType';
import { updateObject } from '../../helpers/utility-helper';
import { IProjectDetailState } from './Types/IProjectDetailState';
import { Notify } from '../../helpers/constants';

const initialState: IProjectDetailState = {
  form: {
    name: 'string',
    contractorId: 1,
    companyId: 1,
    headOfProject: 'string',
    projectOwner: 'string',
    projectManager: 'string',
    pmHasExperience: true,
    scope: 'string',
    cnNumber: 2,
    status: 1,
    engagementId: 2,
    countryId: 3,
    currencyId: 1,
    probabilityOfWinning: 'string',
    approxValue: 'string',
    contractTypeId: 'string',
    cdmNotifiable: true,
    firstAssetWorkedOn: 0,
    secondAssetWorkedOn: 0,
    thirdAssetWorkedOn: 0,
    comment: 'string'
  },
  error: null,
  loading: false,
  notify: Notify.none
};

const projectDetailAddSuccess = (oldState, action) => {
  return updateObject(oldState, {
    error: null,
    loading: false,
    form: action.payload
  });
};

const projectDetailEditSuccess = (oldState, action) => {
  return updateObject(oldState, {
    error: null,
    loading: false,
    form: action.payload
  });
};

const projectDetailError = (oldState, action) => {
  return updateObject(oldState, {
    error: action.error,
    loading: false,
    notify: Notify.error
  });
};

const projectDetailReducer = (oldState = initialState, action) => {
  switch (action.type) {
    case ActionType.PROJECT_ADD:
      return projectDetailAddSuccess(oldState, action);
    case ActionType.PROJECT_EDIT_SUCCESS:
      return projectDetailEditSuccess(oldState, action);
    case ActionType.PROJECT_ADD_ERROR:
      return projectDetailError(oldState, action);
    default:
      return oldState;
  }
};

export default projectDetailReducer;
