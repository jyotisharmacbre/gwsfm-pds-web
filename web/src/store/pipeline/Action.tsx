import * as axios from '../../client';
import { ActionType } from './Types/ActionType';
import { Dispatch } from 'redux';
import IProjectChartSummary from '../../models/IProjectChartSummary';
import IQueryParams from '../../models/tableQueryParams/IQueryParams';
import EventType from '../../enums/EventType';
/* istanbul ignore next */
const projectPipelineDetailSuccess = (response: any) => {
  return {
    type: ActionType.PROJECT_PIPELINE_GRID_DETAILS,
    payload: response
  };
};
/* istanbul ignore next */
const projectPipeineDetailError = (error: string) => {
  return {
    type: ActionType.PROJECT_PIPELINE_GRID_ERROR,
    payload: error
  };
};

/* istanbul ignore next */
const getProjectChartSummarySuccess = (response: Array<IProjectChartSummary>) => {
  return {
    type: ActionType.GET_PROJECT_CHART_SUMMARY_SUCCESS,
    payload: response
  };
};
/* istanbul ignore next */
const getProjectChartSummaryError = (error: string) => {
  return {
    type: ActionType.GET_PROJECT_CHART_SUMMARY_ERROR,
    payload: error
  };
};

const setloadingTrue = () => {
  return {
    type: ActionType.SET_LOADING_TRUE,
  };
};

const headers = {
  'Content-Type': 'application/json'
};

export const projectPipelineDetail = (queryParams: IQueryParams) => {
  return (dispatch: Dispatch) => {
    dispatch(setloadingTrue());
    axios.baseAPI
      .post('api/Projects/GetAll', queryParams, { headers: headers })
      .then(response => {
        /* istanbul ignore next */
        dispatch(projectPipelineDetailSuccess(response.data));
      })
      .catch(error => {
        /* istanbul ignore next */
        dispatch(projectPipeineDetailError(error));
      });
  };
};

export const getProjectChartSummary = () => {
  return (dispatch: Dispatch) => {
    axios.baseAPI
      .get('api/Projects/GetProjectChartSummary', { headers: headers })
      .then(response => {
        /* istanbul ignore next */
        dispatch(getProjectChartSummarySuccess(response.data));
      })
      .catch(error => {
        /* istanbul ignore next */
        dispatch(getProjectChartSummaryError(error));
      });
  };
};
