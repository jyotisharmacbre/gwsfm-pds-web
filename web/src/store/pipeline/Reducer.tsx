import { ActionType } from './Types/ActionType';
import { updateObject } from '../../helpers/utility-helper';
import { IProjectPipelineGridState } from './Types/IProjectPipelineGridState';
import moment from 'moment';
import Notify from '../../enums/Notify';
export const initialState: IProjectPipelineGridState = {
    projectChartSummary: {
        data: [],
        loading: false,
        error: null
    },
    totalNumberOfRecord: 1,
    data: [
        {
            projectId: '',
            projectRefId: 0,
            name: '',
            headOfProject: '',
            contractorId: '-1',
            probabilityOfWinning: 1,
            lastModified: moment().toString(),
            status: '',
            commenceDate: '',
            approxValue: 0,
            contractTypeId: 0,
            cdmNotifiable: false,
            soldMargin: '',
            projectStatusDescription: '',
            contractTypeDescription: ''
        }],
    error: null,
    loading: false
};

const projectPipelineDetailSuccess = (oldState, action) => {
    return updateObject(oldState, {
        error: null,
        loading: false,
        data: action.payload?.data,
        totalNumberOfRecord: action.payload?.totalNumberOfRecord,
        pipelineDetails: action.payload
    });
};

const projectPipeineDetailError = (oldState, action) => {
    return updateObject(oldState, {
        error: action.error,
        loading: false,
        notify: Notify.error,
        totalNumberOfRecord: 0,
        pipelineDetails: []
    });
};

const getProjectChartSummarySuccess = (oldState, action) => {
    return updateObject(oldState, {
        projectChartSummary: updateObject(oldState.projectChartSummary, {
            data: action.payload,
            error: null,
            loading: false
        })
    });
};

const getProjectChartSummaryError = (oldState, action) => {
    return updateObject(oldState, {
        projectChartSummary: updateObject(oldState.projectChartSummary, {
            error: action.error,
            loading: false
        })
    });
};

const setloadingTrue = (oldState, action) => {
    return updateObject(oldState, {
        loading: true,
    });
};

const projectPipelineDetailReducer = (oldState = initialState, action) => {
    switch (action.type) {
        case ActionType.PROJECT_PIPELINE_GRID_DETAILS:
            return projectPipelineDetailSuccess(oldState, action);
        case ActionType.PROJECT_PIPELINE_GRID_ERROR:
            return projectPipeineDetailError(oldState, action);
        case ActionType.GET_PROJECT_CHART_SUMMARY_SUCCESS:
            return getProjectChartSummarySuccess(oldState, action);
        case ActionType.GET_PROJECT_CHART_SUMMARY_ERROR:
            return getProjectChartSummaryError(oldState, action);
        case ActionType.SET_LOADING_TRUE:
            return setloadingTrue(oldState, action);
        default:
            return oldState;
    }
};

export default projectPipelineDetailReducer;
