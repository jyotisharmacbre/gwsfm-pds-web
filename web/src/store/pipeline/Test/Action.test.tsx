import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { initialState } from '../Reducer';
import nock from 'nock';
import { baseURL } from '../../../client/client';
import { projectPipelineDetail } from '../Action';
import IQueryParams from '../../../models/tableQueryParams/IQueryParams';
nock(baseURL)
	.get('/api/Projects/GetAll')
	.reply(200, { data: initialState });



const mockStore = configureStore([thunk]);
let reduxStore;
const setUpStore = (initialState) => {
	reduxStore = mockStore({
		project: initialState
	});
};
describe('Pipeline action tests', () => {
	beforeEach(() => {
		setUpStore(initialState);
		reduxStore.clearActions();
	});

	it('should dispatch the projectPipelineDetail', () => {
		setUpStore(initialState);
		let queryParams = {} as IQueryParams;
		reduxStore.dispatch(projectPipelineDetail(queryParams));
		expect(reduxStore.getActions()).toMatchSnapshot();
	});
});
