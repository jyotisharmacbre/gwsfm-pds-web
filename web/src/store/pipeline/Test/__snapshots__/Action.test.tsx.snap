// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pipeline action tests should dispatch the projectPipelineDetail 1`] = `Array []`;
